spanner/pdml.go

```diff
@@ -105,7 +105,7 @@ func (c *Client) partitionedUpdate(ctx context.Context, statement Statement, opt
 func executePdml(ctx context.Context, sh *sessionHandle, req *sppb.ExecuteSqlRequest) (count int64, err error) {
 	var md metadata.MD
 	// Begin transaction.
-	res, err := sh.getClient().BeginTransaction(contextWithOutgoingMetadata(ctx, sh.getMetadata()), &sppb.BeginTransactionRequest{
+	res, err := sh.getClient().BeginTransaction(ctx, &sppb.BeginTransactionRequest{
```

Comment:
Why are we no longer calling `contextWithOutgoingMetadata` here?

Reply:
executePdml is called from partitionedUpdate method where create and pass sessionHandle and request, so it makes sense to set the headers there only, after this change context with metadata is set the parent function itself, it will also save computing `contextWithOutgoingMetadata` multiple times.

Reply:
Ah, makes sense.